src/dotnet/CarbonAware.DataSources.Json/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using CarbonAware.Interfaces;
+using System.Diagnostics;
+
+namespace CarbonAware.DataSources.Json.Configuration;
+
+public static class ServiceCollectionExtensions
+{
+    public static void AddJsonDataSourceService(this IServiceCollection services)
+    {
+        var source = new ActivitySource("JsonDataSource");
+        services.TryAddSingleton<ICarbonIntensityDataSource, JsonDataSource>();
+        services.TryAddSingleton<ActivitySource>(source);
```

Comment:
In the `WattTimeClient` we delayed naming until the activity was started and then use the classname

```
using (var activity = ActivitySource.StartActivity(nameof(WattTimeClient)))
```

Not sure if one is better than the other, more than anything my preference is for consistency.

Reply:
Good point. I was following Robert's pattern without fully understanding the results of that call. I think it's dicey because you would only start one instance of a watt time client, but you can potentially do many instances of a json data source. I removed it for now and figured we can come back to it and flesh out our plan in a telemetry/monitoring story.